content-api.apib

```diff
@@ -0,0 +1,483 @@
+must know the item Id of the News Item you want to retrieve.
+
++ Parameters
+    + _id (required, string, `1`) ... Identifier of the News Item you want to retrieve.
+
++ Response 200 (application/json)
+
+            {
+                "_etag": "abc",
+                "_links": {
+                    "self": {
+                        "href": "/media/1"
+                    }
+                },
+                "abstract": "abstract needed...",
+                "byline": "John Doe",
+                "copyrightholder": "Sourcefabric z.u.",
+                "copyrightnotice": "(c) Copyright Sourcefabric z.u. 2015",
+                "description_text": "Flower picture",
+                "filemeta": {
+                    "Manufacturer": "CANON", "Orientation": "landscape"
+                },
+                "firstcreated": "2013-07-03T10:11:12Z",
+                "headline": "No news today",
+                "language": "en",
+                "mimetype": "image/jpeg",
+                "provider": "My Company LTD",
+                "pubstatus": "usable",
+                "renditions": {
+                    "original": {
+                        "height": 3462,
+                        "href": "http://farm4.staticflickr.com/3665/9203816834_9f62964627_o_d.jpg",
+                        "mimetype": "image/jpeg",
+                        "width": 5198
+                    },
+                    "thumbnail": {
+                        "height": 150,
+                        "href": "http://farm4.staticflickr.com/3665/9203816834_3329fac058_q_d.jpg",
+                        "mimetype": "image/jpeg",
+                        "width": 150
+                    }
+                },
+                "type": "picture",
+                "urgency": 4,
+                "uri": "tag:mycomp-ltd.com,0000:newsml_LM1EA5F0UUD01",
+                "usageterms": "You are not allowed.",
+                "version": "1",
+                "versioncreated": "2013-07-03T10:11:12Z"
+            }
+
+## News Item Collection [/items?q={q}&type={type}&date_range={date_range}&fields={fields}&limit={limit}]
+
+### List News Items [GET]
+
+This method is used to retrieve a list of News Items.
+
++ Parameters
+    + q (optional, object) ... Elasticsearch query
+    + type (optional, string) ... Specifies the content type for the news items to be
+      retrieved. Valid content types are: text, image, video, audio and composite. Default
+      value: All available media types will be retrieved.
+    + date_range (optional, date interval) ... The date range for the news items to be
+      retrieved. If no end date is specified, the assumed end date will be the current time.
+    + fields (optional, list) ... The list of fields associated with content to be retrieved.
+      Default value: All available fields will be retrieved.
+    + limit (optional, number, `25`) ... The max. number of items to be retrieved.
```

Comment:
I would also use a default soft limit, e.g. 50 or 100 so that by default one cannot accidentally pull a huge amount of data (up to the hard limit, which might be quite high).